components/view_image/view_image.jsx

```diff
@@ -196,13 +201,45 @@ export default class ViewImageModal extends React.PureComponent {
     }
 
     onMouseEnterImage = () => {
+        const fileInfo = this.props.fileInfos[this.state.imageIndex];
+
         this.setState({showCloseBtn: true});
+
+        if (fileInfo && fileInfo.extension && fileInfo.extension === FileTypes.PDF) {
+            this.setState({showZoomBtn: true});
+        }
     }
 
     onMouseLeaveImage = () => {
+        const fileInfo = this.props.fileInfos[this.state.imageIndex];
+
         this.setState({showCloseBtn: false});
+
+        if (fileInfo && fileInfo.extension && fileInfo.extension === FileTypes.PDF) {
+            this.setState({showZoomBtn: false});
+        }
     }
 
+    handleZoomIn = () => {
+        let newScale = this.state.scale;
+        newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
```

Comment:
I initially pulled this from a repository that implemented some zoom functionality for PDF JS. The more I look at how they’re handling the scale factor, the less I understand why they made something so simple, so complicated. I’m going to refactor this to simply:

```
const newScale = Math.min(this.state.scale + DEFAULT_SCALE DELTA), MAX_SCALE);
```